apps/dashboard/src/components/workflow-editor/steps/step-drawer.tsx

```diff
@@ -46,7 +46,18 @@ export const StepDrawer = ({ children, title }: { children: React.ReactNode; tit
           transition={transitionSetting}
         />
         <SheetPortal>
-          <SheetContentBase asChild onInteractOutside={handleCloseSheet} onEscapeKeyDown={handleCloseSheet}>
+          <SheetContentBase
+            asChild
+            onInteractOutside={(e) => {
+              // IMPORTANT: do not close the sheet if the interact outside is from the maily variable list
+              if (e.target instanceof HTMLDivElement && e.target.className.includes('tippy-box')) {
+                return;
+              }
```

Comment:
The problem: Maily renders the variables dropdown with `tippy` which renders it outside the normal tree flow and at the bottom inside the body. Clicking inside of the list is treated as clicking outside the sidebar, which results in closing the sidebar.
I wasn't able to find a better solution to this.